### PR TITLE
haproxy: Allow DNS changes

### PR DESCRIPTION
**Pull request by kklin**
Before, HAproxy resolved hostnames once at startup, and kept the same
association for the lifetime of the service, even if the IP changed in
DNS. This is obviously incorrect when load balanced containers get
restarted with new IPs. HAproxy now tries to updates IPs when health
checks to the old IP fail.

### CODE REVIEW

File: specs/haproxy/config/haproxy.cfg [hunk 1]
```diff
@@ -7,6 +7,9 @@ defaults
     timeout client 5000
     timeout server 5000
 
+resolvers dns
+    nameserver gateway 10.0.0.1:53
```
Comment (ejj) [inline specs/haproxy/config/haproxy.cfg]: I was under the impression that this wasn't necessary and it could use /etc/resolv.conf instead?

Comment (kklin) [inline specs/haproxy/config/haproxy.cfg]: Oh, that'd be nice. I can't find any documentation on that online though, could you point me to it?

Comment (kklin) [inline specs/haproxy/config/haproxy.cfg]: ping @ejj

Comment (ejj) [inline specs/haproxy/config/haproxy.cfg]: Yeah I think I just misinterpreted the documentation.  I think I'd give it a go once without that line, but if it doesn't work we'll just hardcode it for now.

Comment (ejj) [inline specs/haproxy/config/haproxy.cfg]: Yeah that was my thinking.

**Comment by kklin** [comment]
Going to submit this PR on the new quilt/haproxy repo.

**Pull request closed without merging**